### PR TITLE
Updating For Ruby 3.1

### PR DESCRIPTION
**Pull request by johnmccrae**
Signed-off-by: John McCrae <john.mccrae@progress.com>

Updated RUBY_PLATFORM calls pull in the new ucrt from Ruby 3.1

### CODE REVIEW

**Review by marcparadise — APPROVED** [review]

File: .github/workflows/func_spec.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-2019, windows-2016]
+        os: [windows-2019, windows-2022]
         # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
         ruby: [2.7, '3.0']
```
Comment (PrajaktaPurohit) [inline .github/workflows/func_spec.yml:17]: We are not yet testing this on Ruby 3.1, correct?

File: chef-utils/lib/chef-utils/dsl/platform_family.rb [hunk 2]
```diff
@@ -275,7 +275,7 @@ def windows?(node = __getnode(true))
       # @return [Boolean]
       #
       def windows_ruby?
-        !!(RUBY_PLATFORM =~ /mswin|mingw32|windows/)
+        !!(RUBY_PLATFORM =~ /mswin|mingw|windows/)
```
Comment (PrajaktaPurohit) [inline chef-utils/lib/chef-utils/dsl/platform_family.rb:278]: can we include a comment related to why we are doing this?

**Review by PrajaktaPurohit — APPROVED** [review]

File: .github/workflows/func_spec.yml (hunk 1 above)
Comment (johnmccrae) [reply .github/workflows/func_spec.yml:17]: Correct, working towards that, need to get these changes in first

File: .github/workflows/func_spec.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-2019, windows-2016]
+        os: [windows-2019, windows-2022]
```
[the remark this answers is not in the document]
Comment (johnmccrae) [reply .github/workflows/func_spec.yml:15]: Windows Server 2016 is no longer supported as a test platform in Github Workflows as of the 1st of this month.

**Comment by jasonwbarnett** [comment]
@johnmccrae Any reason not to use rubygems abstraction to determine whether or not it's a windows platform? see: https://github.com/rubygems/rubygems/blob/19e32079040bc42ba1716d4a2f644fdceddb86f9/lib/rubygems.rb#L993-L1000

For long term support it may make sense to use an abstraction so this sort of thing doesn't need to happen again. Whether it's the Rubygem's abstraction or something specific for Chef.

**Pull request merged**